distribution/lib/Standard/Base/0.0.0-dev/src/Enso_Cloud/Enso_File.enso

```diff
@@ -38,6 +38,7 @@ from project.Data.Text.Extensions import all
 from project.Enso_Cloud.Public_Utils import get_required_field
 from project.System.File_Format import Auto_Detect, Bytes, File_Format, Plain_Text_Format
 from project.System.File.Generic.File_Write_Strategy import generic_copy
+from project.System.File import find_extension_from_name
 
 type Enso_File
```

Comment:
I was hoping to find here an explanation why there is `Enso_File`, when we already have `System.File`? But no, I don't see the reason.

Reply:
Fair, I will add some comment.

In very short - `Enso_File` is a reference to a file on the Enso Cloud, it is a sibling to e.g. `S3_File`.

Reply:
It seems to me as a copy of `System.File` created because you guys wanted to abstract something. However there already is [FileSystem abstraction](https://www.graalvm.org/truffle/javadoc/org/graalvm/polyglot/io/FileSystem.html) built into Truffle and it is fully accessible from `System.File`.

Reply:
We've been considering to do this at some point, but for the first iterations it seemed like an overkill. 

This PR is _not introducing _ `Enso_File` anyway, it is merely refactoring how it works internally - so if we want to migrate to the `FileSystem` based approach, let's create a separate ticket.

Still, I'm not 100% sure it is something we need right now, especially as while it may work OK with `Enso_File`, I'm not sure if it will work with the sibling `S3_File` - there we also hold credentials corresponding to a given S3 bucket access, and I'm not sure if the Truffle `FileSystem` API would allow us to do that?

Reply:
Tracked in #9600

Reply:
> hold credentials corresponding to a given S3 bucket access

An example of virtualization is a [memory-based file system](https://github.com/oracle/graal/blob/master/truffle/src/com.oracle.truffle.api.test/src/com/oracle/truffle/api/test/polyglot/MemoryFileSystem.java) and it seems to do some [principal magic](https://github.com/oracle/graal/blob/master/truffle/src/com.oracle.truffle.api.test/src/com/oracle/truffle/api/test/polyglot/MemoryFileSystem.java#L713).